src/core/a11y.js

```diff
@@ -0,0 +1,101 @@
+import { pub } from "./pubsubhub.js";
+import { showInlineWarning } from "./utils.js";
+
+export const name = "core/a11y";
+
+const DISABLED_RULES = [
+  "landmark-one-main", // need to add a <main>, else it marks entire page as errored
+  "landmark-unique",
+  "region",
+  "color-contrast", // too slow 🐢
+];
+
+export async function run(conf) {
+  if (!conf.a11y) {
+    return;
+  }
+
+  const violations = await getViolations(conf.a11y);
+  for (const violation of violations) {
+    /**
+     * We're grouping by failureSummary as it contains hints to fix the issue.
+     * For example, with color-constrast rule, it tells about the present color
+     * contrast and how to fix it. If we don't group, errors will be repetitive.
+     * @type {Map<string, HTMLElement[]>}
+     */
+    const groupedBySummary = new Map();
+    for (const node of violation.nodes) {
+      const { failureSummary, element } = node;
+      const elements =
+        groupedBySummary.get(failureSummary) ||
+        groupedBySummary.set(failureSummary, []).get(failureSummary);
+      elements.push(element);
+    }
+
+    const { id, help, description, helpUrl } = violation;
+    const title = `a11y/${id}: ${help}`;
+    for (const [failureSummary, elements] of groupedBySummary) {
+      const hints = formatHintsAsMarkdown(failureSummary);
+      const details = `\n\n${description}.\n\n${hints}. ([Learn more](${helpUrl}))`;
+      showInlineWarning(elements, title, title, { details });
+    }
+  }
+}
+
+/**
+ * @param {object} opts Options as described at https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#options-parameter
+ * @typedef {{ failureSummary: string, element: HTMLElement }} ViolationNode
+ * @typedef {{ id: string, help: string, helpUrl: string, description: string, nodes: ViolationNode[] }} Violation
+ * @returns {Promise<Violation[]>}
+ */
+async function getViolations(opts) {
+  const options = {
+    rules: Object.fromEntries(
+      DISABLED_RULES.map(id => [id, { enabled: false }])
+    ),
+    ...opts,
+    elementRef: true,
+    resultTypes: ["violations"],
+    reporter: "v1", // v1 includes a `failureSummary`
+  };
+
+  let axe;
+  try {
+    axe = await importAxe();
+  } catch (error) {
+    pub("error", "Failed to load a11y linter.");
```

Comment:
```suggestion
    pub("error", "Failed to load a11y linter. See developer console for details.");
```

Reply:
Heh. I added it and then removed it for no reason.